src/SonarQube.Client/Api/Requests/V5_20/GetQualityProfilesRequest.cs

```diff
@@ -0,0 +1,93 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SonarQube.Client.Models;
+
+namespace SonarQube.Client.Api.Requests.V5_20
+{
+    public class GetQualityProfilesRequest : RequestBase<SonarQubeQualityProfile[]>, IGetQualityProfilesRequest
+    {
+        [JsonProperty("projectKey")]
+        public virtual string ProjectKey { get; set; }
+
+        [JsonProperty("organization")]
+        public virtual string OrganizationKey { get; set; }
+
+        [JsonProperty("defaults")]
+        public virtual bool? Defaults => string.IsNullOrWhiteSpace(ProjectKey) ? (bool?)true : null;
+
+        protected override string Path => "api/qualityprofiles/search";
+
+        public override async Task<SonarQubeQualityProfile[]> InvokeAsync(HttpClient httpClient, CancellationToken token)
+        {
+            var result = await InvokeImplAsync(httpClient, token);
+
+            if (result.StatusCode == HttpStatusCode.NotFound)
+            {
+                // The project has not been scanned yet, get default quality profile
+                ProjectKey = null;
+
+                result = await InvokeImplAsync(httpClient, token);
+            }
+
+            result.EnsureSuccess();
+
+            return result.Value;
+        }
+
+        protected override SonarQubeQualityProfile[] ParseResponse(string response) =>
+            JObject.Parse(response)["profiles"]
+                .ToObject<QualityProfileResponse[]>()
+                .Select(FromResponse)
+                .ToArray();
+
+        private static SonarQubeQualityProfile FromResponse(QualityProfileResponse response) =>
+            new SonarQubeQualityProfile(
+                response.Key, response.Name, response.Language, response.IsDefault, response.LastRuleChange);
+
+        // This class MUST NOT change! If a breaking change in the API is introduced,
+        // create a new versioned request and reimplement the serialization there
```

Comment:
Is there something special about this class, or does the warning apply to all of the data classes?

Reply:
It applies to all data classes. Perhaps I should remove the comment, because it should be copied on all classes, which is inconvenient.